here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/SourceIdHandler.java

```diff
@@ -80,17 +77,31 @@ protected EventProcessingStrategy processingStrategyFor(IEvent event) {
     return event.sendUpstream(request);
   }
 
-  private void transformPropertyOperation(ReadFeatures readRequest) {
+  /**
+   * For the AND case, any property query sub-clause that can be converted into tag query will be converted and returned, leaving the remaining inconvertible clauses intact.
+   * <br>
+   * For the OR case, it is required that every sub-clause must be convertible, else the whole clause will not be converted.
+   * This is because OR relation between types of queries (property, tag, spatial,...) is not supported, only AND is supported and applied at the very end of the request.
+   */
+  public static void mapIntoTagOperation(ReadFeatures readRequest) {
 
     if (readRequest.getQuery().getProperties() == null) {
       return;
     }
 
     IPropertyQuery propertyOp = readRequest.getQuery().getProperties();
 
-    // TODO fix it CASL-710
-    //    PropertyOperationUtil.transformPropertyInPropertyOperationTree(
-    //            propertyOp, SourceIdHandler::mapIntoTagOperation);
+    final Optional<ITagQuery> tagQuery = transformPropertyOperation(propertyOp);
+    if (tagQuery.isPresent()) {
```

Comment:
If you're using `Optional` you can use `ifPresent` with lambda so you won't need to call `get` afterwards :)
```
tagQuery.ifPresent(presentTagQuery -> {
    // ...                                   
});
```

Reply:
Thanks, I will change this place. There is 1 other place I will keep like this, as I need to access the index of the loop.